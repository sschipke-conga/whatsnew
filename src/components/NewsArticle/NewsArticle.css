article {
  background: yellow;
  border-radius: 5px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;

  margin: 10px;

}

a {
  border-radius: 0 0 5px 5px;
  background-color: #B56AF0; 
  color: black;
}

.img-div {
  border-radius: 5px 5px 0 0;
  height: 250px;
  overflow: hidden;
  width: 100%;
}

img {
  width: 100%;
}