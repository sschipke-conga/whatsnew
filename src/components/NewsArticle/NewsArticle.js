import React from 'react';
import './NewsArticle.css';

// NEWSARTICLE COMPONENT CODE GOES HERE
const NewsArticle = ({story}) => (
  <article>
    <div className="img-div">
    <img src={story.img} alt=""/>
    </div>
    <h3>{story.headline}</h3>
    <p>{story.description}</p>
    <a href={story.url} target="blank">Click here for full story</a>
  </article>
)
export default NewsArticle;