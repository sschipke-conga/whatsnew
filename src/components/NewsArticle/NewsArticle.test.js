import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle'

describe('NewsArticle', () => {
  let story = {img:'//www.pic.com', headline: 'EXTRA EXTRA', description:'Read all about it'}
  it('should match a snap shot', () => {
    const wrapper = shallow(<NewsArticle 
    story={story}
    />)
    expect(wrapper).toMatchSnapshot();
  })
});