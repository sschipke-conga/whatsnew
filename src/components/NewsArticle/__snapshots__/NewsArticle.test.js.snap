// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match a snap shot 1`] = `
<article>
  <div
    className="img-div"
  >
    <img
      alt=""
      src="//www.pic.com"
    />
  </div>
  <h3>
    EXTRA EXTRA
  </h3>
  <p>
    Read all about it
  </p>
  <a
    target="blank"
  >
    For full story
  </a>
</article>
`;
